src/App.tsx

```diff
@@ -1,26 +1,61 @@
-/* eslint-disable max-len */
+/* eslint-disable jsx-a11y/label-has-associated-control */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import * as React from 'react';
+import { useState, useEffect } from 'react';
 import { UserWarning } from './UserWarning';
+import { USER_ID } from './api/todos';
+import { getTodos } from './api/todos';
 
-const USER_ID = 0;
+import { Header } from './components/Header/Header';
+import { Main } from './components/Main/Main';
+import { Footer } from './components/Footer/Footer';
+import { Errors } from './components/Errors/Errors';
+import { IsActiveError, IsActiveLink } from './types/types';
+import { TodosContext, ErrorContext } from './utils/Store';
 
 export const App: React.FC = () => {
+  const { todos, setTodos } = React.useContext(TodosContext);
+  const { setIsError } = React.useContext(ErrorContext);
+
+  const [link, setLink] = useState(IsActiveLink.All);
+
+  useEffect(() => {
+    getTodos()
+      .then(setTodos)
+      .catch(() => {
+        setIsError(IsActiveError.Load);
+      });
+  }, [setTodos, setIsError]);
+
+  const visibleTodos = React.useMemo(() => {
+    return todos.filter(t => {
+      if (link === IsActiveLink.Active) {
+        return !t.completed;
+      } else if (link === IsActiveLink.Completed) {
+        return t.completed;
+      }
```

Comment:
do not use else. return immediately result of filtering
```suggestion
      if (link === IsActiveLink.Active) {
        return todos.filter(t => !t.completed);
      }
      
       if (link === IsActiveLink.Completed) {
        return todos.filter(t => t.completed);
      }
      
      return todos;
```